src/openms/source/CHEMISTRY/SvmTheoreticalSpectrumGenerator.cpp

```diff
@@ -743,6 +743,18 @@ namespace OpenMS
 
     Int simulation_type = (Int)param_.getValue("svm_mode");
 
+    if (add_metainfo)
+    {
+      if (spectrum.getIntegerDataArrays().size() == 0)
+      {
```

Comment:
I think it would be somewhat "surprising" if you call a function that is supposed to add this kind of meta-data and then simply decides to not actually add any data